### PR TITLE
[directxmesh] Use CMake build system and add dx12 feature

### PR DESCRIPTION
**Pull request by RT2Code**
Use the provided CMake file instead of the VS solutions to build the library, as recommended in the maintainer guidelines. DirectXMesh can be built without DirectX12 (default for desktop) for Windows 7 Service Pack 1 or later, or with DirectX12 (default for ARM or UWP) for Windows 10.

### CODE REVIEW

**Review by LilyWangL — APPROVED** [review]

File: ports/directxmesh/portfile.cmake [hunk 1]
```diff
@@ -8,68 +8,36 @@ vcpkg_from_github(
...
-elseif (VCPKG_PLATFORM_TOOLSET STREQUAL "v142")
-    set(VS_VERSION "2019")
-else()
-    message(FATAL_ERROR "Unsupported platform toolset.")
-endif()
-
-if(VCPKG_TARGET_IS_UWP)
-    set(SLN_NAME "Windows10_${VS_VERSION}")
+if("dx12" IN_LIST FEATURES OR VCPKG_TARGET_IS_UWP OR TRIPLET_SYSTEM_ARCH STREQUAL "arm64")
+    set(FEATURE_OPTIONS -DBUILD_DX12=ON)
```
Comment (dan-shaw) [inline ports/directxmesh/portfile.cmake:12]: I'm not sure if we want to have different sets of default triplets for different target architectures.

Comment (RT222) [reply ports/directxmesh/portfile.cmake:12]: The dx12 feature is only relevant for Windows desktop. You cannot build DirectXMesh without it on ARM or UWP, that's why I force it on on these platforms.

File: ports/directxmesh/CONTROL [hunk 2]
```diff
@@ -1,4 +1,8 @@
 Source: directxmesh
 Version: aug2020
+Port-Version: 1
 Homepage: https://walbourn.github.io/directxmesh
-Description: DirectXMesh geometry processing library
\ No newline at end of file
+Description: DirectXMesh geometry processing library
```
Comment (strega-nil) [inline ports/directxmesh/CONTROL:5]:
```suggestion
Build-Depends: directxmesh[core,dx12](arm64|uwp)
```

File: ports/directxmesh/portfile.cmake [hunk 3]
```diff
@@ -8,68 +8,36 @@ vcpkg_from_github(
...
-        string(REPLACE "_WIN32_WINNT=0x0601" "_WIN32_WINNT=0x0A00" _contents "${_contents}")
-        file(WRITE ${SOURCE_PATH}/DirectXMesh/DirectXMesh_${SLN_NAME}.vcxproj "${_contents}")
-        
-        # fix solution file to include DirectX 12 in build
-        file(READ ${SOURCE_PATH}/Meshconvert/Meshconvert_${SLN_NAME}.vcxproj _contents)
-        string(REPLACE "_WIN32_WINNT=0x0601" "_WIN32_WINNT=0x0A00" _contents "${_contents}")
-        file(WRITE ${SOURCE_PATH}/Meshconvert/Meshconvert_${SLN_NAME}.vcxproj "${_contents}")
-    endif()
+    set(FEATURE_OPTIONS -DBUILD_DX12=OFF)
 endif()
```
Comment (strega-nil) [inline ports/directxmesh/portfile.cmake:15]:
```suggestion
vcpkg_check_features(
    OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    FEATURES
        dx12 BUILD_DX12
)
```

File: ports/directxmesh/portfile.cmake [hunk 4]
```diff
@@ -8,68 +8,36 @@ vcpkg_from_github(
...
+    PREFER_NINJA
+    OPTIONS ${FEATURE_OPTIONS}
 )
 
-file(INSTALL
-    ${SOURCE_PATH}/DirectXMesh/DirectXMesh.h
-    ${SOURCE_PATH}/DirectXMesh/DirectXMesh.inl
-    DESTINATION ${CURRENT_PACKAGES_DIR}/include
-)
+if(NOT VCPKG_TARGET_IS_UWP AND NOT TRIPLET_SYSTEM_ARCH STREQUAL "arm64")
```
Comment (strega-nil) [inline ports/directxmesh/portfile.cmake:23]:
```suggestion
if(NOT VCPKG_TARGET_IS_UWP AND NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
```

File: ports/directxmesh/portfile.cmake [hunk 5]
```diff
@@ -8,68 +8,36 @@ vcpkg_from_github(
...
-    ${SOURCE_PATH}/DirectXMesh/Bin/${SLN_NAME}/${BUILD_ARCH}/Release/DirectXMesh.pdb
-    DESTINATION ${CURRENT_PACKAGES_DIR}/lib)
+file(INSTALL ${SOURCE_PATH}/DirectXMesh/DirectXMesh.h DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+file(INSTALL ${SOURCE_PATH}/DirectXMesh/DirectXMesh.inl DESTINATION ${CURRENT_PACKAGES_DIR}/include)
 
+file(GLOB_RECURSE DEBUG_LIB ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/bin/CMake/*.lib)
+file(GLOB_RECURSE RELEASE_LIB ${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-rel/bin/CMake/*.lib)
+file(INSTALL ${DEBUG_LIB} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib)
+file(INSTALL ${RELEASE_LIB} DESTINATION ${CURRENT_PACKAGES_DIR}/lib)
 if(NOT VCPKG_TARGET_IS_UWP AND NOT TRIPLET_SYSTEM_ARCH STREQUAL "arm64")
```
Comment (strega-nil) [inline ports/directxmesh/portfile.cmake:36]:
```suggestion
if(NOT VCPKG_TARGET_IS_UWP AND NOT VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
```

**Review by strega-nil — CHANGES_REQUESTED** [review]
This is probably a better way to do this.

**Comment by RT222** [comment]
I didn't know it was possible to specify the library own features as dependencies, this is way cleaner. Thanks for your help.

File: ports/directxmesh/portfile.cmake (hunk 1 above)
Comment (RT222) [reply ports/directxmesh/portfile.cmake]: strega-nil fixed this. :)

**Comment by strega-nil** [comment]
@RT222 yeah, course! it's pretty weird so I'm not surprised you wouldn't've thought of it :)

**Review by strega-nil — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**